editor/tests/org.fusesource.ide.projecttemplates.tests.integration/src/org/fusesource/ide/projecttemplates/tests/integration/wizards/FuseIntegrationProjectCreatorRunnableIT.java

```diff
@@ -254,30 +260,58 @@ private IEditorPart getCurrentActiveEditor() {
 		return PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getActiveEditor();
 	}
 
-	private void launchDebug(IProject project) throws InterruptedException {
-		final ExecutePomAction executePomAction = new ExecutePomAction();
+	protected void launchDebug(IProject project) throws InterruptedException, IOException, MalformedObjectNameException, DebugException {
+		final File parent = new File("target/MavenLaunchOutputs");
+		parent.mkdirs();
+		final String mavenOutputFilePath = new File(parent, "MavenLaunchOutput-"+project.getName()+".txt").getAbsolutePath();
+		final ExecutePomAction executePomAction = new ExecutePomAction(){
+			
+			@Override
+			protected void appendAttributes(IContainer basedir, ILaunchConfigurationWorkingCopy workingCopy, String goal) {
+				super.appendAttributes(basedir, workingCopy, goal);
+				System.out.println("Maven output file path: "+mavenOutputFilePath);
+				workingCopy.setAttribute("org.eclipse.debug.ui.ATTR_CAPTURE_IN_FILE", mavenOutputFilePath);
+			}
+			
+		};
 
 		executePomAction.setPostProcessor(new ExecutePomActionPostProcessor() {
 
 			@Override
 			public void executeOnSuccess() {
-				// TODO: shutdown
-				deploymentFinished = true;
-				isDeploymentOk = true;
+				//Won't happen
 			}
 
 			@Override
 			public void executeOnFailure() {
+				//if there is a Maven build failure
 				deploymentFinished = true;
 				isDeploymentOk = false;
 			}
 		});
 		executePomAction.launch(new StructuredSelection(project), ILaunchManager.DEBUG_MODE);
 		int currentAwaitedTime = 0;
 		while (currentAwaitedTime < 30000 && !deploymentFinished) {
-			Thread.sleep(100);
-			currentAwaitedTime += 100;
+			readAndDispatch(0);
+			try{
+				JMXConnector jmxc = JMXConnectorFactory.connect(new JMXServiceURL(ICamelDebugConstants.DEFAULT_JMX_URI));
+				MBeanServerConnection mbsc = jmxc.getMBeanServerConnection();
+				deploymentFinished = !mbsc.queryMBeans(new ObjectName(CamelDebugFacade.CAMEL_DEBUGGER_MBEAN_DEFAULT), null).isEmpty();
+				isDeploymentOk = deploymentFinished;
+				System.out.println("JMX connection succeeded");
+				System.out.println("isDeployment Finished? " + isDeploymentOk);
+				jmxc.close();
+			} catch(IOException ioe){
+				System.out.println("JMX connection attempt failed");
+			}
+			Thread.sleep(500);
+			currentAwaitedTime += 500;
 		}
-		assertThat(isDeploymentOk).isTrue();
+		assertThat(isDeploymentOk).as("build/deployment failed, you can have a look to the file "+ mavenOutputFilePath + " for more information.").isTrue();
+		launch = executePomAction.getLaunch();
+		assertThat(Stream.of(launch.getDebugTargets())

```

Comment:
Understand.  In general though, there may be more than one place in a single line of code that could be the source of an NPE, so having an explicit assertion can save a little debugging time.  Obviously, not the case here, but just considering future maintainability.

> On Jul 25, 2016, at 11:03 AM, Aurélien Pupier <notifications@github.com> wrote:
> 
> In editor/tests/org.fusesource.ide.projecttemplates.tests.integration/src/org/fusesource/ide/projecttemplates/tests/integration/wizards/FuseIntegrationProjectCreatorRunnableIT.java <https://github.com/fusesource/fuseide/pull/599#discussion_r72092542>:
> 
> >  		}
> > -		assertThat(isDeploymentOk).isTrue();
> > +		assertThat(isDeploymentOk).as("build/deployment failed, you can have a look to the file "+ mavenOutputFilePath + " for more information.").isTrue();
> > +		launch = executePomAction.getLaunch();
> > +		assertThat(Stream.of(launch.getDebugTargets())
> having a NPE in tests is fine, it allows simpler code and fails the test anyway
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/fusesource/fuseide/pull/599/files/8b5824e21069f639a4a4e0b8457b1a7df13efc4c#r72092542>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAarXaVagQ8cRyLHHv8rNy6eFWwxKeV3ks5qZN5ZgaJpZM4JS3jw>.
>